crates/iceberg/testdata/view_metadata/ViewMetadataV2Valid.json

```diff
@@ -0,0 +1,58 @@
+{
```

Comment:
it would be good to have more JSON files for testing (not just the happy case)

Reply:
also in general I think this needs more extensive testing. For example on the java side we have https://github.com/apache/iceberg/tree/5ea78e3fbe5d8c9c846a1b86bcd2b77d13a31acd/core/src/test/java/org/apache/iceberg/view.

Specifically in `TestViewMetadata` we have a bunch of test cases that would be good to cover here as well

Reply:
Ok, thanks for the feedback @nastra. I'll add some more tests next week.

Reply:
@nastra I added more tests that cover non-happy cases. 
However, there isn't much in https://github.com/apache/iceberg/tree/5ea78e3fbe5d8c9c846a1b86bcd2b77d13a31acd/core/src/test/java/org/apache/iceberg/view we can use for now. The `ViewMetadataBuilder` I started with is on the same level as the `TableMetadataBuilder` - it can do almost nothing except changing the UUID. The functions tested in the java part are simply not implemented yet. We do have a more complete implementation over at our Iceberg catalog and want to PR them here (https://github.com/hansetag/iceberg-catalog/issues/46), I just want to start with the foundation before heading into the more controversial builder bits.
Would this be alright for you?

Besides the Metadata-Builder, I would like to create 2 follow-up issues:
- Currently in `Snapshots` we use a fallible method for the timestamp() method. For views I am using a non-fallible version, creating a small inconsistency. I would create a new issues to remove `unwrap` from timestamps where they are used.
- Allow for empty namespaces, following our discussion below. I am not sure exactly how to handle this, but right now Rust does not allow the creation of the empty Namespace as used in spark (https://github.com/apache/iceberg-rust/blob/48f9e3e8bff0224569da8dd5beb89ed0c8cc4513/crates/iceberg/src/catalog/mod.rs#L111).

Reply:
> However, there isn't much in https://github.com/apache/iceberg/tree/5ea78e3fbe5d8c9c846a1b86bcd2b77d13a31acd/core/src/test/java/org/apache/iceberg/view we can use for now. 

In the java code we typically don't construct the full JSON just to test something, but there are plenty of tests in `TestViewMetadata` to make sure schema / view version re-assignment / de-duplication happens or SQL dialect deduplication behaves correctly. 

> I just want to start with the foundation before heading into the more controversial builder bits.

I was merely asked to look at this view implementation since I did the Java implementation, but I'm not involved in iceberg-rust, so I'll let others decide how to proceed with adding views to iceberg-rust and what to include in this PR vs in follow-ups. 
Regarding follow-ups, I think a few more things would be needed for the view metadata builder, which I mentioned above. `TestViewMetadata´ gives a good overview over how the view metadata / builder should behave.

Thanks for working on this btw 💯

Reply:
Ok, thanks Eduard for the Feedback.

My preferred way of going forward would be:

1. Merge this PR which just contains the structs & (de)serialization (very much like the `TableMetadaData` in its current state)
2. Figure out a good way for the `TableMetadataBuilder` including partition binding. We have a first shot ready that we can create a PR for.
3. Use the same pattern we use for the `TableMetadataBuilder` for the much lighter `ViewMetadataBuilder` which then includes all the tests Eduard mentioned.

I am aware that a lot of things are missing in the builder. It was a deliberate decision from me to get views up-to-speed with tables first, and then in a second step extend both tables and views features to the java level.

@ZENOTME , @nastra would that be OK for you?